src/lib/tray.js

```diff
@@ -0,0 +1,166 @@
+      visible: false
+    },
+    {
+      id: 'startIpfs',
+      label: i18n.t('start'),
+      click: () => { ipcMain.emit('startIpfs') },
+      visible: false
+    },
+    {
+      id: 'stopIpfs',
+      label: i18n.t('stop'),
+      click: () => { ipcMain.emit('stopIpfs') },
+      visible: false
+    },
+    { type: 'separator' },
+    {
+      label: i18n.t('status'),
+      click: () => { launchWebUI('/') }
+    },
+    {
+      label: i18n.t('files'),
+      click: () => { launchWebUI('/files') }
+    },
+    {
+      label: i18n.t('settings'),
+      click: () => { launchWebUI('/settings') }
+    },
+    { type: 'separator' },
+    {
+      label: i18n.t('advanced'),
+      submenu: [
+        {
+          label: i18n.t('openLogsDir'),
+          click: () => { shell.openItem(app.getPath('userData')) }
+        },
+        {
+          label: i18n.t('openRepoDir'),
+          click: () => { shell.openItem(store.get('ipfsConfig.path')) }
+        },
+        {
+          label: i18n.t('openConfigFile'),
+          click: () => { shell.openItem(store.path) }
+        }
+      ]
+    },
+    {
+      label: i18n.t('about'),
+      submenu: [
+        {
+          label: i18n.t('versions'),
+          enabled: false
+        },
+        {
+          label: `ipfs-desktop ${require('../../package.json').version}`,
+          click: () => { shell.openExternal('https://github.com/ipfs-shipyard/ipfs-desktop/releases') }
+        },
+        {
+          label: `go-ipfs ${require('../../package.json').dependencies['go-ipfs-dep']}`,
+          click: () => { shell.openExternal('https://github.com/ipfs/go-ipfs/releases') }
+        },
+        { type: 'separator' },
+        {
+          label: i18n.t('checkForUpdates'),
+          click: () => { checkForUpdates() }
+        },
+        {
+          label: i18n.t('viewOnGitHub'),
+          click: () => { shell.openExternal('https://github.com/ipfs-shipyard/ipfs-desktop/blob/master/README.md') }
+        }
+      ]
+    },
+    {
+      label: i18n.t('quit'),
+      click: () => { app.quit() }
+    }
+  ])
+}
+
+function icon (color) {
+  const p = path.resolve(path.join(__dirname, '../icons/tray'))
+
+  if (os.platform() === 'darwin') {
+    return path.join(p, `${color}.png`)
+  }
+
+  return path.join(p, `${color}-big.png`)
+}
+
+export default function (ctx) {
+  logger.info('[tray] starting')
+  const tray = new Tray(icon('black'))
+  let menu = null
+  let status = {}
+
+  if (os.platform() !== 'darwin') {
+    // Show the context menu on left click on other
+    // platforms than macOS.
+    tray.on('click', event => {
+      event.preventDefault()
+      tray.popUpContextMenu()
+    })
+  }
+
+  const setupMenu = () => {
+    menu = buildMenu(ctx)
+    tray.setContextMenu(menu)
+    tray.setToolTip('IPFS Desktop')
+    updateStatus(status)
+  }
+
+  const updateStatus = data => {
+    status = data
+
+    menu.getMenuItemById('ipfsIsStarting').visible = status.starting && !status.done
```

Comment:
How about just setting the status to one of these string values in the first place? with the collection of booleans, it's harder to be sure that we have covered every eventuality. If status can only be 1 string, then we can quickly scan the deamon init code to see all the possible values,  and we can just use it directly here.

Reply:
Yes, I wanted to make those constants and just compare. I'll do that in another PR.

Reply:
it would be even better to just change the text and the icon... But this is not native and Electron doesn't let us do that.